go/service/user_report.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2019 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package service
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/keybase/client/go/chat/globals"
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/protocol/chat1"
+	gregor1 "github.com/keybase/client/go/protocol/gregor1"
+	keybase1 "github.com/keybase/client/go/protocol/keybase1"
+	"golang.org/x/net/context"
+)
+
+type convTranscript struct {
+	Messages []convTranscriptMsg `json:"messages"`
+}
+
+type convTranscriptMsg struct {
+	SenderUsername string            `json:"senderUsername"`
+	Body           chat1.MessageBody `json:"body"`
+}
+
+// When sending a transcript, send the following number of most recent chat
+// messages of selected types.
+const transcriptMessageLimit = 100
+
+func pullTranscript(mctx libkb.MetaContext, chatG *globals.ChatContext, convIDStr string) (res string, err error) {
+	convIDBytes, err := chat1.MakeConvID(convIDStr)
+	if err != nil {
+		return "", err
+	}
+	uidBytes := gregor1.UID(mctx.CurrentUID().ToBytes())
+	chatQuery := &chat1.GetThreadQuery{
+		MarkAsRead:   false,
+		MessageTypes: chat1.VisibleChatMessageTypes(),
+	}
+	pagination := &chat1.Pagination{
+		Num: transcriptMessageLimit,
+	}
+	threadView, err := chatG.ConvSource.Pull(mctx.Ctx(), convIDBytes, uidBytes, chat1.GetThreadReason_GENERAL,
```

Comment:
@mmaxim should we have a new `GetThreadReason` for this?

Reply:
Nah, this is going to be a vanishingly small number of the overal reqs